lightning/src/ln/channelmanager.rs

```diff
@@ -5033,22 +5043,23 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 					let raa_updates = break_chan_entry!(self,
 						chan.get_mut().revoke_and_ack(&msg, &self.logger), channel_state, chan);
 					htlcs_to_fail = raa_updates.holding_cell_failed_htlcs;
-					if let Err(e) = self.chain_monitor.update_channel(chan.get().get_funding_txo().unwrap(), raa_updates.monitor_update) {
-						if was_frozen_for_monitor {
-							assert!(raa_updates.commitment_update.is_none());
-							assert!(raa_updates.accepted_htlcs.is_empty());
-							assert!(raa_updates.failed_htlcs.is_empty());
-							assert!(raa_updates.finalized_claimed_htlcs.is_empty());
-							break Err(MsgHandleErrInternal::ignore_no_close("Previous monitor update failure prevented responses to RAA".to_owned()));
-						} else {
-							if let Err(e) = handle_monitor_err!(self, e, channel_state, chan,
-									RAACommitmentOrder::CommitmentFirst, false,
-									raa_updates.commitment_update.is_some(), false,
-									raa_updates.accepted_htlcs, raa_updates.failed_htlcs,
-									raa_updates.finalized_claimed_htlcs) {
-								break Err(e);
-							} else { unreachable!(); }
-						}
+					let update_res = self.chain_monitor.update_channel(chan.get().get_funding_txo().unwrap(), raa_updates.monitor_update);
+					if was_frozen_for_monitor {
+						assert!(update_res != ChannelMonitorUpdateResult::UpdateComplete);
+						assert!(raa_updates.commitment_update.is_none());
+						assert!(raa_updates.accepted_htlcs.is_empty());
+						assert!(raa_updates.failed_htlcs.is_empty());
+						assert!(raa_updates.finalized_claimed_htlcs.is_empty());
+						break Err(MsgHandleErrInternal::ignore_no_close("Previous monitor update failure prevented responses to RAA".to_owned()));
```

Comment:
Not sure if a good way to phrase this, but is there a different word for "failure" when this encompasses "in progress"?